src/generate-grpc-web.ts

```diff
@@ -332,7 +378,20 @@ function createInvokeMethod(ctx: Context) {
               }
             },
           });
-          observer.add(() => client.close());
+          observer.add(() => {
+            if (!observer.closed) return client.close()
+          });
+
+          ${
```

Comment:
These inline ternaries are _just_ long enough to be a little hard to ready; could you pull the out to look like:

```
const maybeAbortSignal = useAbortSignal ? code`
  const abortHandler ...
` ? "";
```

And then inline / like right you can do just `${maybeAbortSignal}`.